src/toil/jobStores/aws/jobStore.py

```diff
@@ -1467,23 +1466,21 @@ def readFrom(self, readable):
                                                       ExtraArgs=headerArgs)
 
                                 # use head_object with the SSE headers to access versionId and content_length attributes
-                                headObj = client.head_object(Bucket=bucket_name,
-                                                             Key=compat_bytes(info.fileID),
-                                                             **headerArgs)
-                                assert dataLength == headObj.get('ContentLength', None)
-                                info.version = headObj.get('VersionId', None)
+                                resp = head_s3_object(
+                                    Bucket=bucket_name,
+                                    Key=compat_bytes(info.fileID),
+                                    **headerArgs
+                                )
+                                assert dataLength == resp.get('ContentLength', None)
+                                info.version = resp.get('VersionId', None)
                                 logger.debug('Upload received version %s', str(info.version))
 
                         if info.version is None:
                             # Somehow we don't know the version
-                            for attempt in retry_s3(predicate=lambda e: retryable_s3_errors(e) or isinstance(e, AssertionError)):
-                                with attempt:
-                                    headObj = client.head_object(Bucket=bucket_name,
-                                                                 Key=compat_bytes(info.fileID),
-                                                                 **headerArgs)
-                                    info.version = headObj.get('VersionId', None)
-                                    logger.warning('Reloaded key with no version and got version %s', str(info.version))
-                                    assert info.version is not None
+                            resp = head_s3_object(Bucket=bucket_name, Key=compat_bytes(info.fileID), header=headerArgs)
+                            info.version = resp.get('VersionId', None)
+                            logger.warning('Reloaded key with no version and got version %s', str(info.version))
```

Comment:
Here's another warning that probably should be either lower-level or conditional.

Reply:
This was code I copy-pasted over in the refactor, but I will make the changes as I agree that that makes sense.